### ISSUE TITLE
problems about net.eval()

### ISSUE THREAD

**Issue by Arieszhang1994** [junyanz/pytorch-CycleGAN-and-pix2pix#285]
Hi junyanz and thank you for your fantastic work!
I am reading your code and found `base_model.eval()`  seems not to be used in your code.
what I have known is just that `base_model.eval()` need to be used when we are testing because batchnorm works differently on this two mode.

I am new to pytorch so I am not sure in the following 2 questions:
1. Do pytorch work in train mode defaultly?
2. Where is the `base_model.eval()` is used? or Did you do other work on changing train/eval mode?

Thank you so much!

**Comment by junyanz** [comment]
For your questions: 
1. Pytorch should work in train mode by default. 
2. Currently `.eval()` is not being used. I added it just in case others would like to use it. It should not affect CycleGAN model at all as CycleGAN has no dropout and batchnorm. It might affect pix2pix model. But we found that pix2pix can produce more diverse results when using dropout during the test time. You are free to try it for your own test code.
